this patch needs to be changed in next release as git already changed

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d53e04..7e3dd2c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,6 +90,10 @@ if(ARCH_ID STREQUAL "ppc64le")
   set(PPC64LE 1)
 endif()
 
+if(ARCH_ID STREQUAL "s390x")
+  set(S390X 1)
+endif()
+
 if(WIN32 OR ARM)
   set(OPT_FLAGS_RELEASE "-O2")
 else()
@@ -451,10 +455,12 @@ else()
 
   option(NO_AES "Explicitly disable AES support" ${NO_AES})
 
-  if(NOT NO_AES AND NOT ARM AND NOT PPC64LE)
+  if(NOT NO_AES AND NOT ARM AND NOT PPC64LE AND NOT S390X)
     message(STATUS "AES support enabled")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -maes")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -maes")
+  elseif(S390X)
+    message(STATUS "AES support not available on s390x")
   elseif(PPC64LE)
     message(STATUS "AES support not available on ppc64le")
   elseif(ARM6)
